share/store/file.go

```diff
@@ -0,0 +1,21 @@
+package store
+
+import (
+	"context"
+	"io"
+
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+type File interface {
+	io.Closer
+	Size() int
+	ShareWithProof(ctx context.Context, axisType rsmt2d.Axis, axisIdx, shrIdx int) (share.Share, nmt.Proof, error)
+	Axis(ctx context.Context, axisType rsmt2d.Axis, axisIdx int) ([]share.Share, error)
```

Comment:
Do we need it? The sampling protocol does not.

Reply:
Removed for now. Could be added later if needed